File: common_book/src/mfxsqbyd/java/com/intelligent/reader/widget/drawer/ShadowView.java
package com.intelligent.reader.widget.drawer;

import android.content.Context;
import android.graphics.Canvas;
import android.graphics.drawable.Drawable;
import android.graphics.drawable.GradientDrawable;
import android.view.View;

/**
 * Desc 滑动返回侧边阴影
 * Author qiantao
 * Mail tao_qian@dingyuegroup.cn
 * Date 2017/11/20
 */

public class ShadowView extends View {

    private Drawable mDrawable;

    public ShadowView(Context context) {
        super(context);
    }

    @Override
    protected void onDraw(Canvas canvas) {
        super.onDraw(canvas);
        if (mDrawable == null) {
            int colors[] = {0x00B8C5D9, 0x12B8C5D9, 0x30B8C5D9};//分别为开始颜色，中间颜色，结束颜色
            mDrawable = new GradientDrawable(GradientDrawable.Orientation.LEFT_RIGHT, colors);
        }
        mDrawable.setBounds(0, 0, getMeasuredWidth(), getMeasuredHeight());
        mDrawable.draw(canvas);
    }
}

File: common_book/src/qbmfrmxs/java/com/intelligent/reader/widget/drawer/ShadowView.kt
package com.intelligent.reader.widget.drawer

import android.content.Context
import android.graphics.Canvas
import android.graphics.Color
import android.graphics.Paint
import android.view.View
import net.lzbook.kit.utils.logd

/**
 * Desc 滑动阴影
 * Author qiantao
 * Mail tao_qian@dingyuegroup.cn
 * Date 2017/11/20
 */

class ShadowView(context: Context) : View(context) {

    private val paint = Paint()

    private var alpha = "00"

    override fun onDraw(canvas: Canvas) {
        super.onDraw(canvas)
        logd("alpha: $alpha")
        paint.color = Color.parseColor("#${alpha}000000")
        paint.style = Paint.Style.FILL
        canvas.drawRect(0f, top.toFloat(), width.toFloat(), bottom.toFloat(), paint)
    }

    fun move(left: Int, right: Int, showing: Float) {
        val hex = 102 - Math.round(showing * 102) // 最大值为 40% 的透明度
        alpha = if (hex < 16) {
            "0" + Integer.toHexString(hex)
        } else {
            Integer.toHexString(hex)
        }
        layout(left, top, right, bottom)
    }
}
